sdk/azcore/runtime/transport_default_http_client.go

```diff
@@ -32,6 +34,12 @@ func init() {
 			Renegotiation: tls.RenegotiateFreelyAsClient,
 		},
 	}
+	if http2Transport, err := http2.ConfigureTransports(defaultTransport); err == nil {
+		// if the connection has been idle for 10 seconds, send a ping frame for a health check
+		http2Transport.ReadIdleTimeout = 10 * time.Second
+		// if there's no response to the ping within 2 seconds, close the connection
+		http2Transport.PingTimeout = 2 * time.Second
```

Comment:
Just mentioning that this assumes that the server supports this. 
Since this defaulted transport is used across SDK for Control Plane and DP services (KV, EventGrid, cosmos etc...) are we sure they are all supporting this ? I assume yes (at least ARM and cosmosdb for sure) but just mentioning it.

Reply:
Also, might out of scope of that PR, but bumping the MaxIddleConnsPerHost from 2 (default) to a bit more is giving some slack to the transport multiplexing. 

Let's say you do only connections to ARM endpoint, then you'll have more than 2 connections handled in the pool that reduce the number of stale connections that can happens.

Reply:
That's a good point about idle connections, I can bump that up a bit.

Agreed on the risk WRT RP support and am trying to perform more live validation to gain confidence.  Worse case, we enable this for the select few clients that have been bumping into this problem.

We're also following up with the KV team about this as our clients have reliably worked for years, so it's a bit odd that this suddenly started happening.

Reply:
For us, this is happening for more than 3 years at random time. It can happens for so many reasons where packet is been dropped : server not able to answer because of load, connections not been gracefully shutdown, NLB dropping connections, Unix kernel, etc... 
Until now, we were forcing http/1 with retries but this is a bummer.

Reply:
Thanks for the context.  It is odd that this suddenly started happening for KV though.

FWIW the `http.Transport` used here invokes the bundled version of `http2.ConfigureTransports()` so the code paths are almost the same, we're just performing this up front and configuring some knobs.

Testing is still in progress.

Reply:
I think it mainly depends how heavy you are using the connection: Eg: You interact with KV DP once a day vs every sec ? The more interaction are happening the more chance you'll have to hit this issue.

Reply:
By not supported, do you mean that the service might not respond to pings?

Reply:
Yep

Reply:
Ugh I see.  So, I suppose we could have a perfectly good connection and if the host doesn't respond to pings, then the transport thinks it's dead and recycles it.  I will follow up with the ARM folks about this to get clarification.  Obviously, this could extend beyond ARM though.  Makes me think that perhaps this should be an opt-in feature.

Reply:
ARM does support H2 ping and "my team" never hit an issue like this on any control plane RP operation. This is mainly happening on any other DP endpoints where ARM is not in the middle. 
I assume ARM does a good job at cleaning connection. Might be interesting to ask them for sure.